Source/DataSources/Angle.js

```diff
@@ -0,0 +1,97 @@
+/*global define*/
+define([
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/DeveloperError',
+        '../Core/Event'
+    ], function(
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError,
+        Event) {
+    "use strict";
+
+    /**
+     * A property type whose value is always interpolated by taking the the shortest distances between 2 angles.
+
+     *
+     * @alias ConstantProperty
+     * @constructor
+     *
+     * @param {Object} [value] The property value.
+     *
+     * @see ConstantPositionProperty
+     *
+     * @exception {DeveloperError} value.clone is a required function.
+     * @exception {DeveloperError} value.equals is a required function.
+     */
+
+    //A value that always interpolates the shortest distances between 2 angles.
+    var Angle = {
+        packedLength : 1,
+        packedInterpolationLength : 1,
+        pack : function(value, array, startingIndex) {
+            startingIndex = defaultValue(startingIndex, 0);
+            array[startingIndex] = value;
+        },
+        unpack : function(array, startingIndex, result) {
+            startingIndex = defaultValue(startingIndex, 0);
+
+            if (!array.length) {
+                result = array;
+            } else {
+                result = array[startingIndex];
+            }
+
+            return result;
+        },
+        convertPackedArrayForInterpolation : function(packedArray, startingIndex, lastIndex, result) {
+            for (var i = 0, len = lastIndex - startingIndex + 1; i < len; i++) {
+                result[i] = packedArray[startingIndex + i];
+            }
+        },
+
+        unpackInterpolationResult : function(value, sourceArray, firstIndex, lastIndex, result) {
+
+        //shortest_angle=((((end - start) % 360) + 540) % 360) - 180;
+        //return shortest_angle * amount;
+
+            var start = sourceArray[firstIndex];
```

Comment:
This solution only works for `LinearApproximation`, it needs to be written to work for all interpolation types in order to be merged into Cesium.  As I discussed on the forum, the correct way to do this (at least the best idea I have right now) would be to modify `convertPackedArrayForInterpolation` so that it stores the list of shortest angles rather than the actual values.  Then you use the interpolated value to find the final rotation in `unpackInterpolationResult`.